codec_array.go

```diff
@@ -59,6 +59,12 @@ func (d *arrayDecoder) Decode(ptr unsafe.Pointer, r *Reader) {
 			r.ReportError("decode array", "size exceeded max allocation size")
 			return
 		}
+
+		if max := r.cfg.getMaxSliceAllocSize(); max > 0 && size > max {
```

Comment:
WDYT of combining this with the previous check. In `getMaxSliceAllocSize` if it is > maxAllocSize, maxAllocSize is returned. This way `MaxSliceAllocSize` defaults to `maxAllocSize`.

Reply:
just changed! Let me know if that's what you expected.

Reply:
Yes, looking good.